apps/web/modules/bookings/views/bookings-single-view.getServerSideProps.tsx

```diff
@@ -66,7 +66,7 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
   if (maybeBookingUidFromSeat.uid) uid = maybeBookingUidFromSeat.uid;
   if (maybeBookingUidFromSeat.seatReferenceUid) seatReferenceUid = maybeBookingUidFromSeat.seatReferenceUid;
 
-  const { bookingInfoRaw, bookingInfo } = await getBookingInfo(uid);
+  const { bookingInfoRaw, bookingInfo, previousBooking } = await getBookingInfo(uid);
```

Comment:
```suggestion
  const { bookingInfoRaw, bookingInfo } = await getBookingInfo(uid);
```

Let's keep this line as it is

Reply:
- In line 79, we have `const rescheduledTo = await BookingRepository.findFirstBookingByReschedule({...`
- `findFirstBookingByReschedule` currently only selects `uid` but you can modify it to return `rescheduledBy` as well
- these are the two fields you need

Reply:
and this will allow you to revert the changes you introduced to some of the other files

Reply:
Agreed, @hbjORbj. However, I found that rescheduled is only true for a meeting that was moved, not for the new meeting that replaces it. I hope this makes sense

so this wont go inside `(bookingInfo.rescheduled)`and wont show the required information on single booking page

if i am correct, either we need to make a call to backend asking for previous booking using fromReschedule which contains the old booking, or we need to get it in the first bookingInfo call inside `bookings-single-view.getServerSideProps`

correct me if i am wrong.

Reply:
![image](https://github.com/user-attachments/assets/d7e815c7-72f1-44fc-9d9d-a17733d54404)

Fetching data from rescheduleTo works for the original meeting, as expected.

Reply:
Hey @hbjORbj , any updates on this?

Reply:
To answer to your comment,

You can create a handler in https://github.com/calcom/cal.com/blob/main/packages/lib/server/repository/booking.ts

and use that in this `getServerSideProps.tsx` file:
e.g.,
```
if (bookingInfo.fromReschedule) {
    const ... = await BookingRepository.findRescheduledByUid({
          uid: bookingInfo.fromReschedule,
        });
}
```

and `findRescheduledByUid` would return:

```
await prisma.booking.findFirst({
      where: {
        uid: uid,
      },
      select: {
        rescheduledBy: true,
        uid: true,
      },
    });
```


Previously, you were querying such a massive data via `getUserBooking` just for these 2 fields, and that's not acceptable since it will hurt our app performance by a lot.